src/unordered.rs

```diff
@@ -952,7 +963,125 @@ impl Plan {
     }
 }
 
-#[cfg(any(test, feature = "serde"))]
+impl MonomialPlan {
+    /// Returns a new FFT plan for the given vector size, selected by the provided `base_n` from a
+    /// compatible [`Plan`].
+    ///
+    /// # Panics
+    ///
+    /// - Panics if `n` is not a power of two.
+    /// - Panics if `base_n` is not a power of two.
+    /// - Panics if `base_n > n`.
+    ///
+    /// # Example
+    /// ```
+    /// use concrete_fft::unordered::MonomialPlan;
+    ///
+    /// let plan = MonomialPlan::new(32, 4);
+    /// ```
+    pub fn new(n: usize, base_n: usize) -> Self {
+        assert!(n.is_power_of_two());
+        assert!(base_n.is_power_of_two());
+
+        let nan = c64 {
+            re: f64::NAN,
+            im: f64::NAN,
+        };
+        let mut twiddles = avec![nan; n].into_boxed_slice();
+
+        let theta = -2.0 / n as f64;
+        for (i, twid) in twiddles.iter_mut().enumerate() {
+            let (s, c) = sincospi64(theta * i as f64);
+            *twid = c64 { re: c, im: s };
+        }
+        let mut indices = avec![0usize; n].into_boxed_slice();
+
+        let nbits = n.trailing_zeros();
+        let base_nbits = base_n.trailing_zeros();
+
+        for (i, idx) in indices.iter_mut().enumerate() {
+            *idx = bit_rev_twice_inv(nbits, base_nbits, i);
+        }
+
+        Self {
+            twiddles,
+            base_n,
+            indices,
+        }
+    }
+
+    /// Returns the vector size of the FFT.
+    pub fn fft_size(&self) -> usize {
+        self.twiddles.len()
+    }
+
+    /// Returns the size of the base ordered FFT.
+    pub fn base_n(&self) -> usize {
+        self.base_n
+    }
+
+    /// Performs a forward FFT on the implicit polynomial `X^degree`, storing the result in `buf`.
+    /// The coefficients are permuted in a way that they're compatible with any other FFT of the
+    /// same internal base ordered size.
+    pub fn fwd_monomial(&self, degree: usize, buf: &mut [c64]) {
```

Comment:
how should we proceed for monomials that wrap around on our rings, multiply by -1 on our end or add a case here to manage monomials modulo X^N + 1 ?

Reply:
actually we can create wrapper functions on our end, this primitive is enough to do what we want I think